### PR TITLE
Various lints

### PR DESCRIPTION
**Pull request by Liamsi**
As it is planned to merge iavl into cosmos-sdk, we need to conform to the lints used there:
https://github.com/cosmos/cosmos-sdk/blob/3f438cdbc2ed6b462e3e146403f836c8bb680317/Makefile#L104-L107

This PR contains most changes necessary for this.

It does not contain changes which would silence warnings a la:
`comment on exported method Node.PathToLeaf should be of the form "PathToLeaf ..." (golint)` 
I think in tendermint we do not lint for these, too.

It also merges master into develop (maybe should be a separate change though).

Supersedes #58

### CODE REVIEW

File: proof.go [hunk 1]
```diff
@@ -144,10 +144,14 @@ func (pln proofLeafNode) Hash() []byte {
 // a path to the least item.
 func (node *Node) PathToLeaf(t *Tree, key []byte) (PathToLeaf, *Node, error) {
 	path := new(PathToLeaf)
-	val, err := node._PathToLeaf(t, key, path)
+	val, err := node._pathToLeaf(t, key, path)
```
Comment (melekes) [inline proof.go]: why underscore in the beginning?

File: node.go [hunk 2]
```diff
@@ -531,34 +525,32 @@ func (node *Node) balance(t *Tree) (newSelf *Node, orphaned []*Node) {
...
 	if balance < -1 {
 		if node.getRightNode(t).calcBalance(t) <= 0 {
 			// Right Right Case
 			newNode, orphaned := node.rotateLeft(t)
 			return newNode, []*Node{orphaned}
-		} else {
-			// Right Left Case
-			var rightOrphaned *Node
+		}
+		// Right Left Case
```
Comment (melekes) [inline node.go]: I'd advocate that sometimes `if then else` does make perfect sense and this is one of those cases

```
if left {
} else {
  (right)
}
```

NOT

```
if left {
}

(right)
```

File: proof_range.go [hunk 3]
```diff
@@ -190,9 +188,8 @@ func (proof *RangeProof) verify(root []byte) (err error) {
 	}
 	if !bytes.Equal(rootHash, root) {
 		return cmn.ErrorWrap(ErrInvalidRoot, "root hash doesn't match")
-	} else {
-		proof.rootVerified = true
 	}
+	proof.rootVerified = true
```
Comment (melekes) [inline proof_range.go]: note: this is the good case where `else` is not needed indeed

File: proof_range.go [hunk 4]
```diff
@@ -179,7 +177,7 @@ func (proof *RangeProof) Verify(root []byte) error {
 	return err
 }
 
-func (proof *RangeProof) verify(root []byte) (err error) {
+func (proof *RangeProof) verify(root []byte) (error) {
```
Comment (melekes) [inline proof_range.go]: brackets around `error` can be removed

File: proof.go (hunk 1 above)
Comment (Liamsi) [reply proof.go]: Was there before. I think Jae uses them to mark recursive helpers. I can remove it here but there are many other places.

File: node.go (hunk 2 above)
Comment (Liamsi) [reply node.go]: I kinda agree. It makes more sense here then in many other places. I think the point is that there is a return in the first case (which makes the else obsolete semantically).  Can reintroduce.

**Pull request merged**